src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SafeCryptoHandles.cs

```diff
@@ -182,9 +178,34 @@ internal bool PublicOnly
             }
         }
 
+        internal void SetParent(SafeProvHandle parent)
+        {
+            if (IsInvalid || IsClosed)
+            {
+                return;
+            }
+
+            Debug.Assert(_parent == null);
+            Debug.Assert(!parent.IsClosed);
+            Debug.Assert(!parent.IsInvalid);
+
+            _parent = parent;
+
+            bool ignored = false;
+            parent.DangerousAddRef(ref ignored);
```

Comment:
I'm not sure I follow...

_parent = parent;
GC+Finalize => _parent.DangerousRelease() (refcount = 0) + _parent = null;

Option 1)
_parent.DangerousAddRef() => NullReferenceException

Option 2)
parent.DangerousAddRef() => Isn't there an exception when AddReffing from 0 to 1?

Seems equivalently exceptional.  And if the DangerousRelease didn't set it to 0, we'd still want to set it to 1 so it didn't interfere with other children and imbalance them.  So it sounds to me like the parameter's the safer choice... what am I missing?